frontend/src/app/max-validator.directive.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { Component } from '@angular/core';
+import { async, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { FormsModule, NgForm } from '@angular/forms';
+import { MaxValidatorDirective } from './max-validator.directive';
+
+@Component({
+  selector: 'gl-test',
+  template: `<form>
+    <input type="number" [max]="max" name="value" [(ngModel)]="value"/>
+  </form>`
+})
+class TestComponent {
+  max: number = null;
+  value: number = null;
+}
+
+describe('MaxValidatorDirective', () => {
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [FormsModule],
+      declarations: [TestComponent, MaxValidatorDirective]
+    });
+  }));
+
+  it('should validate', fakeAsync(() => {
+    const fixture = TestBed.createComponent(TestComponent);
+    fixture.detectChanges();
+    tick();
+
+    const form = fixture.debugElement.children[0].injector.get(NgForm);
```

Comment:
nitpicking: I guess that's a copy/paste, but `fixture.debugElement.query(By.directive(NgForm))` would probably work and be more elegant.

Reply:
That's what I tried, but that returns a DebugElement, and I have no dea how to get access to the actual directive instance. componentInstance is null (IIRC). So I copy-pasted the solution from the angular form directive tests.

Reply:
Yeah, I meant to replace `fixture.debugElement.children[0]`. What you tried used to work though, but the `.injector.get(NgForm)` bit is required now... 

In the ebook, when I wrote the tests about forms in a time when there was no docs or exemples, I also expected to have the directive in `componentInstance`. It worked for a while and then was changed in beta.16 (!!) (see https://github.com/angular/angular/issues/8332).

I ended up also using `.references['myForm']` with `<form #myForm='ngForm'>` in some tests. That works too...